### PR TITLE
Component dropdown mega

### PR DESCRIPTION
**Pull request by marcelmoreau**
This is a mega dropdown menu component:
- when hovering  over nav.primary > ul > li, toggle `.open` class on said list item
- `.open` class is chained to `.dropdown .mega`, which transitions a transform to make the dropdown visible
- would love any feedback/improvements to JS/SCSS,whatever

### CODE REVIEW

**Comment by keyesatimarc** [comment]
Looks nice; two thoughts:

Since you're "hiding" the megadrop with a transition, display none seems like overkill and might get dinged by google. Also, it looks like its animating all properties instead of just opacity, so if you change the speed from 50ms to 5 seconds, it drops in from way up high instead of just kinda showing up where it belongs. The 50ms is so fast you can't really see it moving down and appearing, but a slower speed shows what its really up to.

**Comment by marcelmoreau** [comment]
That initial `display: none` ensures that this does not show on small screens. I `block` it at a large breakpoint.

I believe I am only transitioning two properties: `transform` and `opacity`

**Comment by mbfitz** [comment]
The mega dropdown doesn't work in Firefox :(   might be missing some prefixes?

**Comment by marcelmoreau** [comment]
@mbfitz Works on my FF and also Christian's. Do you have an older version?

**Comment by mbfitz** [comment]
It works now - not sure what was going on there. Maybe a cache issue as it worked fine after a hard refresh! Kind of agree with Christian, though. The transition is a little harsh - also - what if it dropped down form the menu and not the top of the page? Seems a little weird to come from the top and not from the parent nav. ns1.com does this, too, in pure css, if you want to take a look. Also, might be nice to add an indication that there's a dropdown menu underneath the top nav, like an arrow or chevron or something.

**Comment by khamer** [comment]
- The transform transition is too fast
- Consider moving the transition onto `.dropdown` from `.dropdown.open` so that it'll use that transition when it closes too
- `.flex` seems like an unnecessary wrapper element to me - do we need this? can we just put display: flex on `.dropdown`? Or, if it's important to have a wrapper, let's not call it `.flex` and imply implementation like that.
- I'm not a big fan of `.item` as a class either - you're not styling to it, I wouldn't include the class in the example. Plain divs should work just fine for the example.
- Don't forget to fill out the comment in _components.scss
- _components.scss:153 - don't think you need the ampersand
- dropdown.plugin.js - this looks solid; the only thing I can pick on is you never use `$navPrimary`, so you don't need that line.

**Comment by marcelmoreau** [comment]
- I could not (quickly) get the dropdown to slide from under the parent `header.primary` element, so it not fades in. Feel free @mbfitz or whomever to modify this
- added dropdown indicator
- renamed the `.flex` and `.item` elements
- added _components comment
- removed extraneous var in the dropdown plugin js

**Comment by marcelmoreau** [comment]
not hovering correctly

**Comment by marcelmoreau** [comment]
ready to review...looking for those plus unos

**Comment by mbfitz** [comment]
No more fade or slide-in effects? Just wanted to point out the transition isn't working - probably because it's display: none to block. If you remove the display properties and try doing visibility:hidden to visible w/ the opacity it should work. Also - you should only need the transition on one, either the .dropdown.mega or .dropdown.mega.open.

**Comment by marcelmoreau** [comment]
Added in an opacity effect. Actually had to use a max-width media query to get this to work correctly.

**Comment by mbfitz** [comment]
one last thing! need to add "visibility: hidden" on .dropdown and then "visibility: visible" on .dropdown.open, otherwise you trigger the dropdown showing up while hovering below the menu.

**Comment by marcelmoreau** [comment]
fixed

**Comment by mbfitz** [comment]
+1!

**Comment by marcelmoreau** [comment]
anyone else?

**Comment by khamer** [comment]
+1

File: js/site.js [hunk 1]
```diff
@@ -1,5 +1,4 @@
 /**
- * TABS
+ * DROPDOWN
  */
-
-$('.tabs').tabs();
\ No newline at end of file
+$('nav.primary').dropdown();
```
Comment (TommyChanthaboune) [inline js/site.js]: Wrap this in a document ready

**Pull request merged**